### DISCUSSION TITLE
Currently not all OAuth grant types are supported

### DISCUSSION THREAD

**Discussion by SuperDJ** [saloonphp/saloon#264] [General]
I was trying to use Saloon with Zoom. But when I try to authenticate as service-to-service I get the following error message:
>The Client ID is empty or has not been provided

Here are the docs for Zoom https://developers.zoom.us/docs/internal-apps/s2s-oauth/#postman.

**Comment by Sammyjo20** [comment]
Hey @SuperDJ!

Please could you provide a little bit more information to help me understand a bit more? What does your connector/requests currently look like? 

Cheers
Sam

**Comment by SuperDJ** [comment]
@Sammyjo20 there currently is no request because I'm unable to make one. But Zoom requires the `grant_type` to be `account_credentials` which currently isn't possible. 

```
POST https://zoom.us/oauth/token?grant_type=account_credentials&account_id={accountId}
HTTP/1.1
Host: zoom.us
Authorization: Basic Base64Encoder(clientId:clientSecret)
```

**Comment by Lirianer** [comment]
Hi @SuperDJ 

From what I understand you could do this using the `use ClientCredentialsGrant` trait and then overriding the `resolveAccessTokenRequest` method with your own request for the Zoom API authentication.

Something like:
```php
protected function resolveAccessTokenRequest(OAuthConfig $oauthConfig, array $scopes = [], string $scopeSeparator = ' '): Request
{
    return new GetZoomAccountCredentialsTokenRequest($oauthConfig, $scopes, $scopeSeparator);
}
```

If there is anything more that you would need to do this authorization flow maybe make a custom authenticator, or something based upon the current `ClientCredentialsGrant` trait.

Hope this helps!

**Comment by SuperDJ** [comment]
@luckykenlin I think that a new issue or discussion but not really related to this issue.

**Comment by Sammyjo20** [comment]
Hey @SuperDJ I'm going to convert this into a discussion where others can discuss
